### PR TITLE
add feature of saving demo inference result

### PR DESCRIPTION
**Pull request by wwdok**
Usage example:

```
# demo image
python demo/demo.py image --config config/EfficientNet-Lite/nanodet-EfficientNet-Lite0_320_special_vehicle_9classes.yml --model workspace/efficient0_320_special_vehicle_9classes/model_best/model_best.pth  --path test/inputs --save_result
# demo video
python demo/demo.py video --config config/EfficientNet-Lite/nanodet-EfficientNet-Lite0_320_special_vehicle_9classes.yml --model workspace/efficient0_320_special_vehicle_9classes/model_best/model_best.pth  --path test/inputs/fire_truck.mp4 --save_result
```
Saving result :
image
image
image

### CODE REVIEW

File: demo/demo.py [hunk 1]
```diff
@@ -94,17 +97,38 @@ def main():
         for image_name in files:
             meta, res = predictor.inference(image_name)
             predictor.visualize(res, meta, cfg.class_names, 0.35)
+            result_image = predictor.visualize(res, meta, cfg.class_names, 0.35)
```
Comment (RangiLyu) [inline demo/demo.py:100]: The predictor.visualize function is called twice here.

**Review by RangiLyu — CHANGES_REQUESTED** [review]
Thanks for your contribution!
The predictor.visualize function is called twice in demo.py L99 and L100. Could you delete one?

**Comment by wwdok** [comment]
> Thanks for your contribution!
> The predictor.visualize function is called twice in demo.py L99 and L100. Could you delete one?

Sorry for the mistake, fix it ~

**Comment by RangiLyu** [comment]
> > Thanks for your contribution!
> > The predictor.visualize function is called twice in demo.py L99 and L100. Could you delete one?
> 
> Sorry for the mistake, fix it ~

Thank you!

**Pull request merged**